src/cost_functions.jl

```diff
@@ -72,34 +72,56 @@ end
 
 struct LogLikeLoss{T,D} <: DECostFunction
   t::T
-  distributions::D
+  data_distributions::D
+  diff_distributions::L where L<:Union{Void,D}
+  weight
 end
+LogLikeLoss(t,data_distributions) = LogLikeLoss(t,data_distributions,nothing,nothing)
+LogLikeLoss(t,data_distributions,diff_distributions) = LogLikeLoss(t,data_distributions,diff_distributions,1)
 
 function (f::LogLikeLoss)(sol::DESolution)
-  distributions = f.distributions
-  fill_length = length(f.t)-length(sol)
-  for i in 1:fill_length
-    push!(sol.u,fill(Inf,size(sol[1])))
-  end
+  if f.diff_distributions == nothing
+    distributions = f.data_distributions
+    fill_length = length(f.t)-length(sol)
+    for i in 1:fill_length
+      push!(sol.u,fill(Inf,size(sol[1])))
+    end
 
-  ll = 0.0
+    ll = 0.0
 
-  if eltype(distributions) <: UnivariateDistribution
-    for j in 1:length(f.t), i in 1:length(sol[1][1])
+    if eltype(distributions) <: UnivariateDistribution
+      for j in 1:length(f.t), i in 1:length(sol[1][1])
       # i is the number of time points
       # j is the size of the system
       # corresponds to distributions[i,j]
-      ll -= logpdf(f.distributions[i,j],sol[i,j])
-    end
-  else # MultivariateDistribution
-    for j in 1:length(f.t), i in 1:length(sol[1][1])
+        ll -= logpdf(distributions[i,j],sol[i,j])
+      end
+    else # MultivariateDistribution
+      for j in 1:length(f.t), i in 1:length(sol[1][1])
       # i is the number of time points
       # j is the size of the system
       # corresponds to distributions[i,j]
-      ll -= logpdf(f.distributions[i],sol[i])
+        ll -= logpdf(distributions[i],sol[i])
+      end
     end
+    ll
+  else
```

Comment:
Since the `if` covers both, it won't put the two together and instead will only use one of the likelihoods.